linalg/complex_densemat.hpp

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_COMPLEX_DENSEMAT
+#define MFEM_COMPLEX_DENSEMAT
+
+#include "complex_operator.hpp"
+#include <complex>
+
+namespace mfem
+{
+
+/** @brief Specialization of the ComplexOperator built from a pair of Dense
+    Matrices.
+    The purpose of this specialization is to support the inverse of a
+    ComplexDenseMatrix and various MatMat operations
+    See ComplexOperator documentation for more information.
+    Note: Only the Hermitian convention is supported
```

Comment:
I have no objection to supporting only this convention at this stage. However, I'd like to see this note expanded. Is there some underlying reason for this? Perhaps something in the LAPACK routines that make this convention more natural to support? Are the potential benefits, if any, of the other convention not worth the extra code?

We don't need to go into great detail but a sentence or two to convince proponents of the other convention that this choice makes sense for dense matrices. Alternatively we could say that the other convention will be supported in the future if the need arises. Either approach is fine with me.

Reply:
I added the block symmetric convention as well. This change only affects the part where we construct the monolithic 2x2 DenseMatrix. The other methods (such as Solve Factor etc) are not affected by the conventions since it's done by using complex arithmetic. In general, I do prefer the Hermitian conventions since if the matrix is actually Hermitian then the 2x2 block is symmetric.